### ISSUE TITLE
GitCommitBear: Add warnings

### ISSUE THREAD

**Issue by virresh** [coala/coala-bears#2311]
The GitCommitBear currently ignores if the remote host that has been added to git is not supported or there is no remote added. 
Add a warning so that the user knows that those options from .coafile essentially are not having any effect.

Example:
Commands to reproduce:
Setup Folders:
```
mkdir gitTest
cd gitTest
```

Setup .coafile and git repository
```
echo "[default]
files = *.txt

[commit]
bears = GitCommitBear
shortlog_trailing_period = False
body_close_issue = True" > .coafile
echo "ShortLog: The commitMessage.

No issue closed." > commitMsg
git init
git add commitMsg .coafile
git commit -F commitMsg
```

Run coala
```
coala
```

What happens:
There is a warning regarding shortlog having a fullstop. But no warning regarding absence of host. The ```body_close_issue=True``` in .coafile essentially is having no effect. The user is not being warned.

Suggested corrections:
Probably add the required checks in GitCommitBear.py and update corresponding tests in GitCommitBearTest.py

Also note that no warning is generated in case the remote host is not supported  (```the one listed by git remote -v```). This also need to be checked for.

**Comment by jayvdb** [comment]
ew, this is problematic because it depends on the order of the remotes in `.git/config`, or the remote names, and power users (all coala devs, right?) should be creating their own enhanced workflows based on remote names.

We do not yet have a good 'get upstream/central-repo remote' algorithm.  That needs to be solved first before there is a warning.  However we could have an 'info' message.

**Comment by virresh** [comment]
Yes, true. We are currently just picking up the first remote from the .git/config, so we don't know for sure which is the intended central repo, and the problem worsens in case the same local repo is corresponding to multiple remotes...
Probably we can have the issue reference flag also take in some part of the actual upstream repository ?
(Something like issueReference = github.com/upstream_repo_name in our coala config and a fallback to the first remote in case it's absent )

**Comment by virresh** [comment]
Can I work on this and add warnings for incompatible hosts and no remotes ?
(Something similar to what is done in HgCommitBear ...)
